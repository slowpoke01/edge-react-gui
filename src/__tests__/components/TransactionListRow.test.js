/* globals describe it expect */
/* eslint-disable flowtype/require-valid-file-annotation */

import * as React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import { getTheme } from '../../components/services/ThemeContext.js'
import { TransactionListRowComponent } from '../../components/themed/TransactionListRow.js'

describe('Request', () => {
  it('should render with loading props', () => {
    const renderer = new ShallowRenderer()

    const props = {
      cryptoAmount: 'string',
      denominationSymbol: 'string',
      fiatAmount: 'string',
      fiatSymbol: 'string',
      isSentTransaction: true,
      requiredConfirmations: 11,
      selectedCurrencyName: 'string',
      thumbnailPath: 'string',
      walletBlockHeight: 11,
      // eslint-disable-next-line react/no-unused-prop-types
      walletId: 'string',
      // eslint-disable-next-line react/no-unused-prop-types
      currencyCode: 'string',
      transaction: 'string' | 11,
      theme: getTheme()
    }
    const actual = renderer.render(<TransactionListRowComponent {...props} />)

    expect(actual).toMatchSnapshot()
  })
})
