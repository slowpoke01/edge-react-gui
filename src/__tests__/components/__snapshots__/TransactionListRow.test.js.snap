// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request should render with loading props 1`] = `
<WithTheme(TransactionRowComponent)
  cryptoAmount="string"
  denominationSymbol="string"
  fiatAmount="string"
  fiatSymbol="string"
  isSentTransaction={true}
  onPress={[Function]}
  requiredConfirmations={11}
  selectedCurrencyName="string"
  thumbnailPath="string"
  transaction={11}
  walletBlockHeight={11}
/>
`;
